compiler/codegen/cg-expr.cpp

```diff
@@ -4720,8 +4720,37 @@ static GenRet codegenGPUKernelLaunch(CallExpr* call, bool is3d) {
   std::vector<GenRet> args;
   int curArg = 1;
   for_actuals(actual, call) {
-    if (curArg > nNonKernelParamArgs) {
-      args.push_back(codegenAddrOf(actual));
+    Symbol* actualSym = toSymExpr(actual)->symbol();
+    if (curArg == 1) {
+      if (FnSymbol* fn = toFnSymbol(actualSym)) {
+        args.push_back(new_CStringSymbol(fn->cname));
+      }
+      else if (actualSym->isImmediate()) {
+        args.push_back(actual->codegen());
+      }
+      else {
+        INT_FATAL("Unknown argument type in GPU launch primitive");
+      }
+    }
+    else if (curArg > nNonKernelParamArgs) {
```

Comment:
Looks like for each argument we now pass the argument itself along with a size. Given that this is the case can you update the comment on line on line 4711 (tangent: I wish Github let me attach this review comment to that line), which for now says:

```C++
  // The runtime function needs the kernel parameters to be passed by address.
  // The other arguments to this primitive are passed along directly.
```